test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/SignedAttributeTests.cs

```diff
@@ -0,0 +1,278 @@
+#if NET46
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using FluentAssertions;
+using Moq;
+using NuGet.Common;
+using NuGet.Packaging.Signing;
+using NuGet.Packaging.Signing.DerEncoding;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class SignedAttributeTests
+    {
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithAuthorSignature()
+        {
+            var attribute = AttributeUtility.GetCommitmentTypeIndication(SignatureType.Author);
+            AttributeUtility.GetCommitmentTypeIndication(attribute).Should().Be(SignatureType.Author);
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithRepositorySignature()
+        {
+            var attribute = AttributeUtility.GetCommitmentTypeIndication(SignatureType.Repository);
+            AttributeUtility.GetCommitmentTypeIndication(attribute).Should().Be(SignatureType.Repository);
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithAuthorSignatureAndUnknownVerifyAuthor()
+        {
+            var attribute = GetCommitmentTypeTestAttribute(Oids.CommitmentTypeIdentifierProofOfOrigin, "1.3.6.1.5.5.7.3.3");
+            AttributeUtility.GetCommitmentTypeIndication(attribute).Should().Be(SignatureType.Author);
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithUnknownSignatureType()
+        {
+            var attribute = GetCommitmentTypeTestAttribute("1.3.6.1.5.5.7.3.3");
+            AttributeUtility.GetCommitmentTypeIndication(attribute).Should().Be(SignatureType.Unknown);
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithBothRepoAndAuthorVerifyFailure()
+        {
+            var attribute = GetCommitmentTypeTestAttribute(Oids.CommitmentTypeIdentifierProofOfReceipt, Oids.CommitmentTypeIdentifierProofOfOrigin);
+            AttributeUtility.IsValidCommitmentTypeIndication(attribute).Should().BeFalse();
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithDuplicateValuesVerifyFailure()
+        {
+            var attribute = GetCommitmentTypeTestAttribute(Oids.CommitmentTypeIdentifierProofOfOrigin, Oids.CommitmentTypeIdentifierProofOfOrigin);
+            AttributeUtility.IsValidCommitmentTypeIndication(attribute).Should().BeFalse();
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithNoValueVerifyFailure()
+        {
+            var attribute = GetCommitmentTypeTestAttribute();
+            AttributeUtility.IsValidCommitmentTypeIndication(attribute).Should().BeFalse();
+        }
+
+        [Fact]
+        public void SignedAttribute_CommitmentTypeIndicationWithBothRepoAndAuthorVerifyThrows()
```

Comment:
It would be great if the test name contained the member under test (`GetCommitmentTypeIndication` in this case).  This test is very similar to the test on line 51 above except for the member under test.  You can eliminate `SignedAttribute` from the test name as I don't think it adds much value.  Test reporting prints the full name.

Reply:
I'll update the names to include the member.

Yes, the tests between the Get which throws, and the IsValid are very similar. The Get isn't used yet since we aren't consuming this property beyond validating that it isn't an incorrect combination so I've made sure that it has extra tests.